data/onCreatePage.ts

```diff
@@ -11,17 +15,49 @@ const pageLayoutOptions: Record<string, LayoutOptions> = {
   '/docs/404': { sidebar: false, searchBar: false, template: '404' },
 };
 
-export const onCreatePage: GatsbyNode['onCreatePage'] = ({ page, actions }) => {
-  const { createPage } = actions;
+// Function to extract code element classes from an MDX file
+const extractCodeLanguages = async (filePath: string): Promise<Set<string>> => {
```

Comment:
A first go at something a little less "runtime". Since the syntax for defining codeblocks is set by Markdown which is quite rigid itself - we can dig through MDX files for present languages and build a representative superset that way. Textile files are still determined at runtime as they are still tied into the half-Textile-half-React world.